pkg/daemon/ptpdev.go

```diff
@@ -28,3 +33,38 @@ func GetDevStatusUpdate(nodePTPDev *ptpv1.NodePtpDevice) (*ptpv1.NodePtpDevice,
 	}
 	return nodePTPDev, nil
 }
+
+func runDeviceStatusUpdate(ptpClient *ptpclient.Clientset,nodeName string) {
+	// Discover PTP capable devices
+	// Don't return in case of discover failure
+	ptpDevs, err := ptpnetwork.DiscoverPTPDevices()
+	if err != nil {
+		glog.Errorf("discover PTP devices failed: %v", err)
+	}
+	glog.Infof("PTP capable NICs: %v", ptpDevs)
+
+	// Assume NodePtpDevice CR for this particular node
+	// is already created manually or by PTP-Operator.
+	ptpDev, err := ptpClient.PtpV1().NodePtpDevices(PtpNamespace).Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		glog.Errorf("failed to get NodePtpDevice CR for node %s: %v", nodeName, err)
+	}
+
+	// Render status of NodePtpDevice CR by inspecting PTP capability of node network devices
+	ptpDev, _ = GetDevStatusUpdate(ptpDev)
+
+	// Update NodePtpDevice CR
+	_, err = ptpClient.PtpV1().NodePtpDevices(PtpNamespace).UpdateStatus(ptpDev)
```

Comment:
Does it make sense to update the status only if it's different from the current one?

Reply:
there is room for improvements here for sure but I will like to do it on a different PR if that is fine by you